spec/Section 5 -- Validation.md

```diff
@@ -570,6 +577,50 @@ fragment conflictingDifferingResponses on Pet {
 }
 ```
 
+Fragment spread arguments can also cause fields to fail to merge.
+
+While the following is valid:
+
+```graphql example
+fragment commandFragment($command: DogCommand!) on Dog {
+  doesKnowCommand(dogCommand: $command)
+}
+
+fragment potentiallyConflictingArguments(
+  $commandOne: DogCommand!
+  $commandTwo: DogCommand!
+) on Dog {
+  ...commandFragment(command: $commandOne)
+  ...commandFragment(command: $commandTwo)
+}
```

Comment:
I think it's good to allow for `...commandFragment ...commandFragment` (two argumentless fragment-spreads) for backwards compatability and `...commandFragment(a: $a) ...commandFragment(a: $a)` because they are the same which in my mind ties intuitively into the former case while the different arguments should be disallowed for ease of reasoning otherwise we'd have to go into the fragment-spreads and analyse whether this changed variable results in conflicting fields due to arguments on the field or `@include/@skip`/...

Reply:
I agree with Matt - this should use the same logic as field usage. This would thus be invalid for the same reason that this query (and a query using any two of these three fields) would be invalid:

```graphql
query ($id1: ID! = "ZmlsbXM6MQ==", $id2: ID! = "ZmlsbXM6MQ==") {
  node(id: "ZmlsbXM6MQ==") { id }
  node(id: $id1) { id }
  node(id: $id2) { id }
}
```

[Example on SWAPI](http://graphql.github.io/swapi-graphql/?query=query%20(%24id1%3A%20ID!%20%3D%20%22ZmlsbXM6MQ%3D%3D%22%2C%20%24id2%3A%20ID!%20%3D%20%22ZmlsbXM6MQ%3D%3D%22)%20%7B%0A%20%20node(id%3A%20%22ZmlsbXM6MQ%3D%3D%22)%20%7B%20id%20%7D%0A%20%20node(id%3A%20%24id1)%20%7B%20id%20%7D%0A%20%20node(id%3A%20%24id2)%20%7B%20id%20%7D%0A%7D)

Reply:
Made it more clear in https://github.com/graphql/graphql-spec/pull/1081/commits/03ba25598f7520c05b3dfb27697a84de76829c2e